src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -1591,6 +1592,31 @@ public void validateDeleteRequest(Dataset dataset, List<DataDeletionTableModel>
         }
     }
 
+    /*
+     * WARNING: Ensure SQL is validated before executing this method!
+     */
+    public List<Map<String, Object>> getSnapshotTableData(Snapshot snapshot,
+                                                          String sql) throws InterruptedException {
+        // execute query and get result
+        final BigQueryProject bigQueryProject = bigQueryProjectForSnapshot(snapshot);
+        final TableResult result = bigQueryProject.query(sql);
+
+        // aggregate into single object
+        final FieldList columns = result.getSchema().getFields();
```

Comment:
it might be good to factor this out as a general `sqlResultToElasticMapping` utility function.

Reply:
That makes sense, although it's not SQL, it's BigQuery `TableResult`. I think since it's ES specific it could go in an ES utils class, or a BQ utils class.

Reply:
What would be the goal in separating this method from the `BigQueryPdao` class?